tcmur_cmd_handler.c

```diff
@@ -253,11 +242,24 @@ static void handle_unmap_cbk(struct tcmu_device *dev, struct tcmulib_cmd *ucmd,
 	error = state->error;
 	pthread_mutex_unlock(&state->lock);
 
-	unmap_state_free(state);
+	pthread_mutex_destroy(&state->lock);
+	free(state);
 
 	aio_command_finish(dev, origcmd, error ? status : ret);
```

Comment:
I think if you make the 2 other changes then state->status will always hold the correct value and we can just do

aio_command_finish(dev, origcmd, status);